src/main/scala/zio/s3/MultipartUploadOptions.scala

```diff
@@ -0,0 +1,35 @@
+package zio.s3
+
+import software.amazon.awssdk.services.s3.model.CreateMultipartUploadRequest
+import zio.s3.MultipartUploadOptions.MinMultipartPartSize
+
+import scala.jdk.CollectionConverters._
+
+/**
+ * The upload options that are specific to multipart uploads
+ *
+ * @param uploadOptions [[UploadOptions]]
+ * @param partSize the size of the part in bytes, the minimum is 5 MB
+ * @param parallelism the number of parallel requests to upload chunks, default to 1
+ */
+case class MultipartUploadOptions(
+  uploadOptions: UploadOptions = UploadOptions.default,
```

Comment:
it is usually not a good practice to set default value in case class, better to do it companion object